### PR TITLE
Respect ACL for multi-value custom field

### PR DESCRIPTION
**Pull request by pradpnayak**
Overview
----------------------------------------
ACL is not respected on multi-value custom field

Before
----------------------------------------
Can perform ADD/Update/Copy/Delete actions on VIEW Only permission

After
----------------------------------------
Cannot perform ADD/Update/Copy/Delete actions on VIEW Only permission

### CODE REVIEW

File: CRM/Profile/Page/MultipleRecordFieldsListing.php [hunk 1]
```diff
@@ -449,6 +455,13 @@ public function browse() {
         }
       }
     }
+
+    if (!CRM_Core_BAO_CustomGroup::isCustomGroupAllowed(
```
Comment (MegaphoneJon) [inline CRM/Profile/Page/MultipleRecordFieldsListing.php]: This PR works well overall, but the logic on this `if` statement is inverted.

Comment (pradpnayak) [reply CRM/Profile/Page/MultipleRecordFieldsListing.php]: Hi @MegaphoneJon 

Good spot. Have just pushed fix.

Thanks for reviewing and doing QA.
Pradeep

**Comment by colemanw** [comment]
@pradpnayak most of this looks to be user-facing. How are acls enforced for these fields at the bao/api layer?

File: templates/CRM/Profile/Page/MultipleRecordFieldsListing.tpl [hunk 2]
```diff
@@ -114,7 +114,7 @@
     <div id='{$dialogId}' class="hiddenElement"></div>
   {/if}
 
-  {if !$reachedMax}
+  {if !$reachedMax && $showAddButton}
```
Comment (seamuslee001) [inline templates/CRM/Profile/Page/MultipleRecordFieldsListing.tpl]: could this create an e-notice if the showAddButton is not assigned at all?

**Comment by jaapjansma** [comment]
Betty and I are reviewing PR's and we are doing the least recently updated first. So we came across this one. 

We have two comments:
* We are missing a description of how to reproduce this issue. See below of what we figured out.
* This PR cannot be merged anymore. @pradpnayak are you planning to fix this if not we would advice to close this PR
* We have not tested this PR as it could not be merged.

**Steps to reproduce**

*We reproduced this in drupal*
1. Add a new user with a limited role
2. Make sure the role **doesn't** have the permissions: edit all contacts, administer civicrm, access all custom data
3. Create a new group called Access Group (type Access Control)
4. Under Administer --> Permissions --> Manage Roles --> add a new role
5. Under Administer --> Permissions --> Assign Users --> Assign the group to the role
6. Under Administer --> Permissions --> Manage ACL --> Add a new rule: select role the created role, operation: view, type of data a set of custom fields, custom data: select the custom data the user is allowed to view but not edit
7. Add the newly created user in step 1 to the group Access Group
8. Login as the newly created user and view a contact
9. Check the data on the custom group tab and try to add or edit. 

**Expected result in step 9**
Not possible to add or edit existing data

**Actual result in step 9**
Possible to add or edit data.

**Comment by jaapjansma** [comment]
We will close this PR.

**Review by dwoods-encircle — COMMENTED** [review]
Tested manually with 5.29.0 and works well.

**Comment by dwoods-encircle** [comment]
@jaapjansma  is there an alternative PR for this functionality? We have a requirement for this in a live deployment and would be good to get this into next release if possible (We are merging files with each release). Let me know if we can help.

**Comment by MegaphoneJon** [comment]
@dwoods-encircle Jaap didn't write this PR - and it wouldn't go into the next release regardless, as an RC has been cut.  It was closed because it needed rebasing, which didn't happen.  If you have a rebased version, it would be good to submit that as a PR.

**Comment by jaapjansma** [comment]
Indeed I did not write this PR I only reviewed it and closed as it no one responded to the review. So feel free to reopen this PR again or to provide a new one with the feedback as with the reviewing I could still add/edit data whilst I wouldn't expect that.

**Comment by encircle** [comment]
@jaapjansma apologies, I've rebased the changes against this new PR: https://github.com/civicrm/civicrm-core/pull/19026

**Pull request closed without merging**